airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/bump_version/pipeline.py

```diff
@@ -124,9 +119,9 @@ async def _run(self) -> StepResult:
                 stdout="Can't retrieve the connector current version.",
                 output_artifact=self.repo_dir,
             )
-        updated_metadata = self.get_metadata_with_bumped_version(current_version, self.new_version, current_metadata)
-        repo_dir_with_updated_metadata = metadata_change_helpers.get_repo_dir_with_updated_metadata(
-            self.repo_dir, metadata_path, updated_metadata
+        updated_metadata_str = self.get_metadata_with_bumped_version(current_version, self.new_version, current_metadata_str)
```

Comment:
I totally get the reason why you mutate the metadata as string and not yaml: preserving order and comment. 
I would suggest adding a  `yaml.safe_load(current_metadata_step)` step without assignation to make sure the yaml is still valid.
Or even better: [run a `MetadataValidation` step.](https://github.com/airbytehq/airbyte-private/blob/master/airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/metadata/pipeline.py#L24).

FYI  I think this [python yaml package](https://pypi.org/project/ruamel.yaml/) can preserve comment and order, but I never tried it